Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/Cancellation/CancelProgressFlowNode.cs

```diff
@@ -212,5 +212,89 @@ private bool CheckIfWillComplete(bool isStillProcessing, ref EntityProxyInstance
                 }
             }
         }
+
+        /// <summary>
+        /// A wrapper that provides access to data version(s) shared between <see cref="ITaskSetOwner"/> instances.
+        /// This allows multiple <see cref="CancelProgressFlowNode"/>s that point to the same <see cref="ITaskSetOwner"/>
+        /// instance to track whether data for a given version has been processed.
+        ///
+        /// Call <see cref="Dispose()"/> on the instance when it is no longer required to release shared version
+        /// references.
+        /// </summary>
+        /// <remarks>
+        /// Example:
+        /// Multiple <see cref="CancelProgressFlowNode"/>s point to each <see cref="TaskDriverSystem{TTaskDriverType}"/>
+        /// instance.
+        /// </remarks>
+        private struct OwnerSharedDataVersion : IDisposable
+        {
+            private VersionRefWrapper m_Ref;
+
+            public uint ProgressLookupDataVersion
+            {
+                get => m_Ref.ProgressLookupDataVersion;
+                set => m_Ref.ProgressLookupDataVersion = value;
+            }
+
+
+            public OwnerSharedDataVersion(ITaskSetOwner owner)
+            {
+                m_Ref = VersionRefWrapper.Acquire(owner);
+            }
+
+            public void Dispose()
+            {
+                m_Ref.Release();
```

Comment:
```suggestion
                m_Ref?.Release();
```

If we call this twice, we'll get an error.